cartoframes/layer.py

```diff
@@ -305,22 +325,53 @@ def _setup(self, layers, layer_idx):
                 self.scheme = mint(5)
             elif self.style_cols[self.color] in ('date', 'geometry', ):
                 raise ValueError('Cannot style column `{col}` of type '
-                                 '`{type}`. It must be numeric, string, or '
+                                 '`{type}`. It must be numeric, text, or '
                                  'boolean.'.format(
                                      col=self.color,
                                      type=self.style_cols[self.color]))
 
-        # if color not specified, choose a default
-        self.color = self.color or DEFAULT_COLORS[layer_idx]
-        self.cartocss = self._get_cartocss(basemap)
-
         if self.time:
+            # don't use turbo-carto for animated maps
             column = self.time['column']
             frames = self.time['frames']
             method = self.time['method']
             duration = self.time['duration']
-            agg_func = "'{method}({time_column})'".format(method=method,
-                                                          time_column=column)
+            if (self.color in self.style_cols and
+                    self.style_cols[self.color] in ('string', 'boolean', )):
+                self.query = ' '.join([s.strip() for s in [
+                    'SELECT',
+                    '    orig.*, __wrap.cf_value_{col}',
+                    'FROM ({query}) AS orig, (',
+                    '    SELECT',
+                    '      row_number() OVER (',
+                    '        ORDER BY val_{col}_cnt DESC) AS cf_value_{col},',
+                    '      {col}',
+                    '    FROM (',
+                    '        SELECT {col}, count({col}) AS val_{col}_cnt',
+                    '        FROM ({query}) as orig',
+                    '        GROUP BY {col}',
+                    '        ORDER BY 2 DESC',
+                    '    ) AS _wrap',
+                    ') AS __wrap',
+                    'WHERE __wrap.{col} = orig.{col}',
+                ]]).format(col=self.color, query=self.query)
+                agg_func = '\'CDB_Math_Mode({})\''.format(
+                        'cf_value_{}'.format(self.color))
+                self.scheme = {
+                        'bins': ','.join(str(i) for i in range(1, 11)),
+                        'name': 'Bold',
```

Comment:
If a user wants to style a animated map by a categorical column, it seems like the only scheme allowed is 'Bold'. If a user could runs the following code, no styling change happens.
```Python
cc.map(layers=[BaseMap(),
               Layer('all_month_3',
                     time = {'column':'time', 'method':'count', 'trails':5, 'duration':60},
                     color = {'column':'magtype','scheme':styling.pastel(10)})],
       interactive=True)
```

Reply:
Great catch! I need to update that line to
```python
'name': self.scheme.get('name', 'Bold')
```